setup.py

```diff
@@ -17,22 +17,89 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
-
+import functools
 import os
+import shutil
+import subprocess
 import sys
+import tempfile
+from distutils.errors import CompileError
 from pathlib import Path
 
 from setuptools import Extension, find_packages, setup
+from setuptools.command.build_ext import build_ext
+
+ISA_L_SOURCE_ARCHIVE = os.path.join("src", "isal", "isa-l", "v2.30.0.tar.gz")
```

Comment:
I did use a submodule first, but getting a whole directory tree in the package was a nightmare. The tarball was much more straightforward and easy to package. So it is more of a pragmatic than an ideal solution.
I'd rather use the submodule though, but I wonder how I could make that work without making the package a mess :thinking:.

Reply:
Hmm since shutil has an archive method, I can add a submodule of isa-l somewhere outside of the source tree, and package it into a tarball for inclusion in the sdist. Would that be an acceptable compromise?

Reply:
Why was it a nightmare? It could be as easy as adding `graft vendor/isa-l` (or wherever you put the submodule) to `MANIFEST.in`. But I haven’t done this before.

Have a look at https://blog.ionelmc.ro/presentations/packaging/ as well, very useful advice.

Well, putting a tarball into a tarball is ugly, but if it works, I guess it is acceptable.

Reply:
Well I used package data, so I had to use a function with `os.walk` to get all the files. And there are many, many files in the isa-l project. So the whole package tree would be very unbalanced towards a lot of isa-l files with very little python and cython files. 
But yes, if this gets easier with  MANIFEST.in I will use that. I will also check if pyproject.toml has some magic for this. 

> https://blog.ionelmc.ro/presentations/packaging/ 

The guy who learned me packaging (via the blog). I liked his post on the `src` layout so much that I adopted it for all my packages. Will definitely have a look at this.

Reply:
Thanks again for the information. I used a submodule now tagged at v2.30.0. Also I added a manifest.in that grafts the whole submodule tree. In the binary package only the LICENSE, README and Release_notes for ISA-L are included.